### PR TITLE
Fix inconsistent mobile input validation

### PR DESCRIPTION
**Pull request by savindi7**
### Related Issues
- https://github.com/wso2/product-is/issues/13277

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- none

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

Closes wso2/product-is#13277

### CODE REVIEW

File: apps/myaccount/src/extensions/configs/profile.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ProfileConfig } from "./models";
+
+export const profileConfig:  ProfileConfig = {
```
Comment (yasinmiran) [inline apps/myaccount/src/extensions/configs/profile.ts:21]:
```suggestion
export const profileConfig: ProfileConfig = {
```

File: apps/myaccount/src/extensions/configs/profile.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (yasinmiran) [inline apps/myaccount/src/extensions/configs/profile.ts:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```

File: apps/myaccount/src/extensions/configs/models/profile.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.com).
```
Comment (yasinmiran) [inline apps/myaccount/src/extensions/configs/models/profile.ts:2]:
```suggestion
 * Copyright (c) 2022, WSO2 Inc. (http://www.wso2.com).
```

File: apps/myaccount/src/extensions/configs/profile.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ProfileConfig } from "./models";
+
+export const profileConfig:  ProfileConfig = {
+    attributes: {
+        getRegExpValidationError: (attribute: string) => {
+            if(attribute === "phoneNumbers"){
```
Comment (yasinmiran) [inline apps/myaccount/src/extensions/configs/profile.ts:24]:
```suggestion
            if (attribute === "phoneNumbers") {
```

**Review by yasinmiran — APPROVED** [review]

File: apps/myaccount/src/extensions/configs/index.ts [hunk 5]
```diff
@@ -18,3 +18,5 @@
 
 export * from "./common";
 export * from "./account-recovery";
+export * from "./profile";
+
```
Comment (brionmario) [inline apps/myaccount/src/extensions/configs/index.ts:22]:
```suggestion
```

File: apps/myaccount/src/extensions/configs/models/index.ts [hunk 6]
```diff
@@ -18,3 +18,5 @@
 
 export * from "./common";
 export * from "./account-recovery";
+export * from "./profile";
+
```
Comment (brionmario) [inline apps/myaccount/src/extensions/configs/models/index.ts:22]:
```suggestion
```

File: apps/myaccount/src/extensions/configs/profile.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ProfileConfig } from "./models";
+
+export const profileConfig: ProfileConfig = {
+    attributes: {
+        getRegExpValidationError: (attribute: string) => {
+            if (attribute === "phoneNumbers") {
```
Comment (brionmario) [inline apps/myaccount/src/extensions/configs/profile.ts:24]: Get `phoneNumbers` from the contstants.

https://github.com/wso2/identity-apps/blob/69d11407f15f7c7a90861d04ee36f13e2fa946ae/modules/core/src/constants/profile-constants.ts#L62

File: apps/myaccount/src/components/profile/profile.tsx [hunk 8]
```diff
@@ -733,9 +733,13 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): R
                                                                 ));
                                                             } else if (checkSchemaType(schema.name, "phoneNumbers")) {
                                                                 validation.errorMessages.push(t(
-                                                                    "myAccount:components.profile.forms.mobileChangeForm." +
-                                                                "inputs.mobile.validations.invalidFormat"
-                                                                ));
+                                                                    profileConfig?.attributes?.
+                                                                        getRegExpValidationError("phoneNumbers"),
```
Comment (brionmario) [inline apps/myaccount/src/components/profile/profile.tsx:737]:
```suggestion
                                                                        getRegExpValidationError(ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get("PHONE_NUMBERS")), 
```

File: apps/myaccount/src/components/profile/profile.tsx [hunk 9]
```diff
@@ -733,9 +733,13 @@ export const Profile: FunctionComponent<ProfileProps> = (props: ProfileProps): R
                                                                 ));
                                                             } else if (checkSchemaType(schema.name, "phoneNumbers")) {
```
Comment (brionmario) [inline apps/myaccount/src/components/profile/profile.tsx:734]:
```suggestion
                                                            } else if (checkSchemaType(schema.name, ProfileConstants.SCIM2_SCHEMA_DICTIONARY.get("PHONE_NUMBERS"))) {
```

Comment (savindi7) [reply apps/myaccount/src/components/profile/profile.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3043/commits/35ca0bc0741f40aec9fed064394f80d35148e926

File: apps/myaccount/src/components/profile/profile.tsx (hunk 8 above)
Comment (savindi7) [reply apps/myaccount/src/components/profile/profile.tsx]: Fixed with https://github.com/wso2/identity-apps/pull/3043/commits/35ca0bc0741f40aec9fed064394f80d35148e926

File: apps/myaccount/src/extensions/configs/profile.ts (hunk 7 above)
Comment (savindi7) [reply apps/myaccount/src/extensions/configs/profile.ts]: Fixed with https://github.com/wso2/identity-apps/pull/3043/commits/35ca0bc0741f40aec9fed064394f80d35148e926

File: apps/myaccount/src/extensions/configs/models/index.ts (hunk 6 above)
Comment (savindi7) [reply apps/myaccount/src/extensions/configs/models/index.ts]: Fixed with https://github.com/wso2/identity-apps/pull/3043/commits/35ca0bc0741f40aec9fed064394f80d35148e926

File: apps/myaccount/src/extensions/configs/index.ts (hunk 5 above)
Comment (savindi7) [reply apps/myaccount/src/extensions/configs/index.ts]: Fixed with https://github.com/wso2/identity-apps/pull/3043/commits/35ca0bc0741f40aec9fed064394f80d35148e926

**Review by brionmario — APPROVED** [review]

**Pull request merged**